### PR TITLE
feat(react): add support for react 18

### LINKED ISSUE #15: Support React 18
React 18 is now generally available. At the moment the `peerDependencies` stipulate:
```json
 "peerDependencies": {
    "react": "^16.8.0 || ^17.0.0",
    "react-dom": "^16.8.0 || ^17.0.0"
  },
```

Could the package be updated to support React 18?
I don't mind creating a PR for this, but I wanted to check that no one else has picked this up already.

### PR DESCRIPTION
**Pull request by afenton90**
This PR adds the following changes:
- **Support for React 18:** `peerDependencies` have been updated to allow `react` & `react-dom` 18 to be installed. Due to the usage of react in this module `16` & `17` can still be supported for now.
- **Storybook upgrade:** Storybook has been upgraded to the latest version to add better support for React 18. There is still one open [issue](https://github.com/storybookjs/storybook/issues/18349) in storybook to add full support, but it works fine without it here.
- **Dependencies updated:**
    - `react-script-hook` required an upgrade to support React 18 so this has been done. I noticed at the same time that this was listed in `devDependencies` when it was used in the main export code. So this has been upgraded into `dependencies` (no changes for module consumers here).
    - `react-select`:  Would have needed an upgrade but in fact isn't used in the module anymore, so has been removed.

closes #15

### CODE REVIEW

**Comment by dmsolutionz** [comment]
👍

File: src/GoCardlessDropinButton.tsx [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
 import { useGoCardlessDropin } from "./hooks";
 import { GoCardlessDropinOptions } from "./types";
 import "./GoCardlessDropinButton.css";
 
 // Opens a Billing Request Flow in a modal. Relies on GoCardlessDropin being
 // loaded into window, which will only happen if the initialise script has been
 // loaded via a script tag.
-export const GoCardlessDropinButton = (options: GoCardlessDropinOptions) => {
-  const { open } = useGoCardlessDropin({ ...options });
+export const GoCardlessDropinButton: FC<GoCardlessDropinOptions> = (props) => {
```
Comment (aniketmalandkar) [inline src/GoCardlessDropinButton.tsx:9]: As per GC coding guidelines, We don't use `FC` to define props type. Rather we define it as it was before.

File: src/index.stories.tsx [hunk 2]
```diff
@@ -45,13 +46,10 @@ const Instructions = () => (
...
 //
-export const Base = ({
-  billingRequestFlowID,
-  environment,
-}: {
+export const Base: FC<{
   billingRequestFlowID: string;
   environment: string;
-}) => {
+}> = ({ billingRequestFlowID, environment }) => {
```
Comment (aniketmalandkar) [inline src/index.stories.tsx:52]: Same here as mentioned in the previous comment.

**Review by aniketmalandkar — APPROVED** [review]
LGTM 👍 Left minor comments. 

Please wait for @oanaOM to review the changes :)

**Review by oanaOM — APPROVED** [review]
Thanks for your contribution on this! 
Besides @aniketmalandkar's comments, LGTM!

File: src/GoCardlessDropinButton.tsx (hunk 1 above)
Comment (afenton90) [reply src/GoCardlessDropinButton.tsx]: All done 👍

File: src/index.stories.tsx (hunk 2 above)
Comment (afenton90) [reply src/index.stories.tsx]: All done 👍

**Comment by aniketmalandkar** [comment]
@afenton90 Thanks for your contribution! :)

**Pull request merged**